_transcripts/050-rust-1.56-1.57.md

```diff
@@ -0,0 +1,1302 @@
+you leak the value, it will never be dropped, and therefore the heap memory is
+just valid forever. So it can give you a `'static` reference to it. And the same
+thing exists for `Vec`, but with `Vec::leak`, what it would do is, it would
+first shrink the allocation to be actually the size of the length of the array,
+not just its capacity, and then give you back a reference to that slice. Of
+course this meant that it was a little weird to call `Vec::leak`, because it did
+an allocation and a copy of the entire vector, in order to move it into the
+smaller capacity. But now with this change, `Vec::leak` will do the sort of
+intuitive thing, of keeping the whole memory allocation and still just giving
+you a reference to the slice of just the contents. Which I thought was
+entertaining, that it didn't do this in the past.
+
+__Ben__: Similarly, the final item in my notes is that, as of the new release,
+Rust now has a new tier 3 target of the Nintendo 3DS.
+
+__Jon__: It's so great. I don't know why— I don't know why this was added, but
+it—
+
+__Ben__: Somebody wanted to write Rust on the 3DS, and they were like, you know
+what, why not? Let's do it. And so we should mention too, that the tier system—
+tier 1 are targets that get, like, you know, tested every single, like every
+single commit gets, like, the full test suite run on it. And so they're very
+well tested. Tier 2 is— they are guaranteed to compile with every commit but
+they were— with every release, that is. But they won't run the full test suite.
+And then tier 3 is anything goes. Like, we'll add support for this if you really
+want us to, but it's up to you to make sure it works. There's no Nintendo 3DS
+test runner somewhere, in someone's, like, kitchen, hooked up the internet,
+with, like, you know, running random Rust code every release to make sure it
+still works.
+
+__Jon__: It's very sad. We should definitely have a fleet of 3DSs.
+
+__Ben__: We should.
+
+__Jon__: I had a couple more things. So in Cargo— and this is actually a change
+that happened in 1.55 but but didn't make the release notes. Which is that Cargo
+will no longer have `RUSTFLAGS` be set for build scripts. It used to be that if
+`RUSTFLAGS` was set in the environment, and then Cargo would just pass that
+through to build scripts. But it wasn't entirely reliable, because if for
+example `RUSTFLAGS` was not set, but there were `rustflags` set in the Cargo
+configuration file through, like, `build.rustflags` for example. Those would not
+be set in `RUSTFLAGS`. So build scripts that relied on the `RUSTFLAGS`,
+environment variable would actually get a sort of misleading value. Now, since
+1.55, and announced now in 1.56, Cargo will set a different environment variable
+called `CARGO_ENCODED_RUSTFLAGS`, which is guaranteed to hold the current set of
+`RUSTFLAGS` that Cargo is using. Encoded in a way where, like, whitespace
+splitting and such isn't a problem, that build scripts can actually rely on.
```

Comment:
The formatting is a little tricky here-- `RUSTFLAGS` is the environment variable, while `rustflags` is the config setting. Please review? (audio 79:05)

Reply:
lgtm :+1: